init_test.go

```diff
@@ -58,17 +58,90 @@ func TestInit(t *testing.T) {
 	}
 }
 
-func funcNameValidation(name string, t *testing.T) {
-	err := commands.ValidateFuncName("fooFunc")
-	if err == nil {
-		t.Error("Expected validation error for function name")
+func TestFuncNameValidation(t *testing.T) {
+	tc := []struct {
+		name        string
+		value       string
+		errExpected bool
+	}{
+		{
+			name:        "strings only",
+			value:       "somename",
+			errExpected: false,
+		},
+		{
+			name:        "numbers only",
+			value:       "12345",
+			errExpected: false,
+		},
+		{
+			name:        "uppercase letters",
+			value:       "HelloFuncName",
+			errExpected: true,
+		},
+		{
+			name:        "starts with separator",
+			value:       ".myfunc",
+			errExpected: true,
+		},
+		{
+			name:        "ends with separator",
+			value:       "myfunc-",
+			errExpected: true,
+		},
+		{
+			name:        "contains spaces",
+			value:       "my func blah",
+			errExpected: true,
+		},
+		{
+			name:        "hypen in name",
+			value:       "my-func-blah",
+			errExpected: false,
+		},
+		{
+			name:        "underscore in name",
+			value:       "my_funcblah",
+			errExpected: false,
+		},
+		{
+			name:        "dot in name",
+			value:       "my.funcblah",
+			errExpected: false,
+		},
+		{
+			name:        "multiple separators in row",
+			value:       "my___funcblah",
+			errExpected: true,
+		},
+		{
+			name:        "multiple different separators in row",
+			value:       "my_.funcblah",
+			errExpected: true,
+		},
+		{
+			name:        "multiple separators in the name",
+			value:       "my_func-blah.test",
+			errExpected: false,
+		},
```

Comment:
there are more cases that suppose to be covered like `.{name}` that evaluates to the following image tag`{registry}/.{name}:{version}` <--- that's not a valid case.

So, i guess, names that start with `*`, `.`, `-` are not valid as well.

Reply:
I have the `.myfunc` test case. I can add the other cases as well.

Reply:
And I am pretty sure there are other cases as well (&, ~@#$%^&*...), but I don't think we actually need to test for all of them :)

Reply:
Added those tests.